### PR TITLE
Update installation procedure to use 2.3.1

### PR DESCRIPTION
**Pull request by tristan0x**
**Description of the Pull Request (PR):**

* Update installation procedure from a released tarball to use the latest version released one week ago. Also factor shell commands to ease maintenance of this page.
* Update installation procedures from a Git clone to use latest tag.

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have tested this PR locally with a `make test`
- [ ] This PR is NOT against the project's master branch
- [ ] I have added myself as a contributor to the [Author's file](https://github.com/singularityware/singularity/blob/master/AUTHORS.md)
- [ ] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by vsoch** [comment]
I might add something about checking out a tag, because given the repo exists, it's not always clear how to do this. For example, I have this branch at remote called "upstream" so I needed to do:

```
git fetch --tags upstream
git checkout tags/2.3.1 -b 2.3.1
```
to get the 2.3.1 tag active in a new branch called 2.3.1

**Comment by tristan0x** [comment]
Thank you @vsoch for this suggestion. I definitely agree that `git checkout tags/<tag> -b <tag>` is much better than `git checkout <tag>`.

The `git fetch...` hint is also very valuable, but I would put it in a dedicated section though: _To compile and install Singularity from an existing Git clone_

WDYT?

**Comment by vsoch** [comment]
+1! And whatever we come up with here, I'll mirror on the site docs too :)

File: INSTALL.md [hunk 1]
```diff
@@ -11,11 +11,12 @@ $ sudo yum groupinstall "Development Tools"
...
 ```
-$ wget "https://github.com/singularityware/singularity/releases/download/2.3/singularity-2.3.tar.gz"
-$ tar -xvzf singularity-2.3.tar.gz
-$ cd singularity-2.3
+$ version=2.3.1
+$ wget "https://github.com/singularityware/singularity/releases/download/${version}/singularity-${version}.tar.gz"
+$ tar -xvzf singularity-${version}.tar.gz
+$ cd singularity-${version}
 $ ./configure --prefix=/path/to/singularity
 $ make
```
Comment (vsoch) [inline INSTALL.md]: one more quick comment here! I think given that most don't know the linux filesystem hierarchy, we should give an example like `prefix=/usr/local` so we don't see people installing it in home, Downloads, etc.

File: INSTALL.md [hunk 2]
```diff
@@ -29,6 +30,22 @@ will result in a non-functioning or semi-functioning installation.
...
 $ cd singularity
+$ git checkout tags/2.3.1 -b 2.3.1
+$ ./autogen.sh
+$ ./configure --prefix=/path/to/singularity
+$ make
+$ sudo make install
+```
+
+note: The `sudo` is very important for the `make install`. Failure to do this
+will result in a non-functioning or semi-functioning installation.
```
Comment (vsoch) [inline INSTALL.md:41]: This is **so** important!! Nicely said!

File: INSTALL.md (hunk 1 above)
Comment (tristan0x) [reply INSTALL.md]: I took your remark into account an additional commit.

**Comment by gmkurtzer** [comment]
This looks good to me. Are we ready to merge?

**Comment by vsoch** [comment]
Looks good to me. I would click "update branch" first @gmkurtzer to make sure some new change didn't break this PR (I think since it's install docs it is unlikely). +1 from me!

**Comment by vsoch** [comment]
@gmkurtzer this should be ok to merge, although we need to check whether this should be merged into master (since it's just installation docs) or into development.

**Comment by vsoch** [comment]
hey @tristan0x ! Just a quick poke - this PR needs to be against development branch, then we can merge. Thank you!

**Review by bauerm97 — APPROVED** [review]
Approving this PR on behalf of @gmkurtzer

**Comment by bauerm97** [comment]
Going ahead and merging on behalf of @gmkurtzer

**Pull request merged**